src/sass/checkboxes/_base.scss

```diff
@@ -3,121 +3,112 @@
 
 @use '../core' as *;
 
-.#{$prefix}-checkbox {
-  position: absolute;
-  left: -$spacing-sm;
-  opacity: 0;
-
-  & + .#{$prefix}-label,
-  .#{$prefix}-checkbox-wrapper & ~ .#{$prefix}-label {
-    font-size: $spacing-sm;
-
-    /**
-     * Helps visually align checkbox lists
-     */
-    margin-left: .2rem;
-    font-weight: $font-weight-regular;
-
-    &::before {
-      display: inline-block;
-      content: ' ';
-      border-radius: .125rem;
-      margin-right: .75rem;
-      width: $spacing-sm;
-      height: $spacing-sm;
-      transition: all .2s ease;
-
-      /**
-       * There are some magic numbers here that help improve rendering
-       * of box-shadows in Firefox and IE. Making them just a smidge
-       * wider (.08rem instead of .0625rem/1px) than one pixel compensates
-       * for some weird aliasing that is happening. As a bonus it seem to
-       *  make radios and checkboxes look optically closer the 1px border
-       * used on other inputs like text inputs, textareas, and selects.
-       */
-      box-shadow: 0 0 0 .08rem $color-black-400;
-
-      /**
-       * This helps visually center the label with the pseudo
-       * element checkbox.
-       */
-      vertical-align: middle;
-      margin-top: -.1rem;
-    }
-
-    &:hover {
-      cursor: pointer;
-    }
+.rvt-checkbox {
+  display: inline-block;
+  padding-left: $spacing-md;
+  position: relative;
+
+  input[type=checkbox] {
+    cursor: pointer;
+    height: $spacing-md;
+    left: 0;
+    margin: 0;
+    top: 0;
+    width: $spacing-md;
+    opacity: 0;
+    position: absolute;
   }
 
-  &:checked + .#{$prefix}-label,
-  .#{$prefix}-checkbox-wrapper &:checked ~ .#{$prefix}-label {
-    &::before {
-      color: $color-white-base;
-      background-color: $color-blue-600;
-      box-shadow: 0 0 0 .08rem $color-blue-600;
-      background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgaGVpZ2h0PSIxNiIgdmlld0JveD0iMCAwIDE2IDE2Ij4KICAgIDxwYXRoIGZpbGw9IiNmZmZmZmYiIGQ9Ik02LjcyLDEzbC0uNDgtLjM2LTMtM0ExLDEsMCwwLDEsNC43MSw4LjI5bDIuMTEsMi4xMiw0LjMzLTYuOTRhMSwxLDAsMCwxLDEuNywxLjA2TDcuNjQsMTIuODdaIi8+Cjwvc3ZnPgo=');
-      background-position: 50% 50%;
-      background-repeat: no-repeat;
-    }
+  &--sr-only-label input[type=checkbox] + label,
+  &--sr-only-label input[type=checkbox] ~ label {
```

Comment:
The adjacent sibling connector (`+`) is redundant when also using the general sibling connector (`~`). I'd recommend just using the `~` connector and remove the other, since you're wanting to maintain that option. That would clean several lines of code in this file.

Reply:
That's a good call.